Themis/Views/Assessment/CorrectionSidebar/ProblemStatementCellView.swift

```diff
@@ -70,25 +69,27 @@ struct ProblemStatementCellView: View {
             }
         }
         .padding()
-        .onAppear {
-            vm.convertProblemStatement(
-                problemStatement: problemStatement ?? "",
-                feedbacks: feedbacks,
-                colorScheme: colorScheme)
+        .task(id: submission) {
```

Comment:
No, the submission ID alone is not enough as it alone cannot be observed. If you only pass the submissionid, as above the problemstatement will not be updated when a new submission is loaded.

Reply:
Also it doesn't make any difference wether the submission or the submission id is passed as the comparison function anyways only compares the submission id.

Reply:
Ah, I started reading from bottom to top. You wrote more about this earlier 🤔  I would like to experiment with this a bit more myself as well (also to get a better feeling about when SwiftUI redraws stuff), but I cannot test it right now because the server is down 😅

Reply:
Ah, I started reading from bottom to top. You wrote more about this earlier 🤔  I would like to experiment with this a bit more myself as well (also to get a better feeling about when SwiftUI redraws stuff), but I cannot test it right now because the server is down 😅